### ISSUE TITLE
API/Web Access Logs Requirements

### ISSUE THREAD

**Issue by JasonCTang** [bcgov/SIMS#2790]
***Describe the task***
Changes to logging to satisfy security audit requirements

***Acceptance Criteria***
- [x] Stop logging health check log entries in the API logs (see additional context).
- [x] Write the IP Address and GUID (same saved to the `sims.users.user_name`) of the user associated with any API call log entries.
  - [x] Current IP address can be found in header `http_x_forwarded_for`. Similar log was also done for #18.
  - [x] Review with the team the final suggested log format.

**Additional context**

- Current API health check logs.

<img width="718" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBNmNWQnc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--7e63b7ceb1772cacf8eebc5b64821dc29d02e5dd/image.png" alt="image.png" />

- David Malcolm has confirmed that we are allowed to write the user guids to the logs.

**Comment by sslaws** [comment]
I've confirmed that web was already recording the x_forwarded_for and I updated the configuration to filter out the health checks.  The API needs to read the x_forwarded_for and write it and the guid of the user to the associated logs.

**Comment by andrewsignori-aot** [comment]
@sslaws I would say the first AC below would be no longer needed after your last commit, right? During the call we were able to see the client IP correctly logged. Would you agree to remove the below AC?

_Make necessary changes to the web app access logs to write the actual address of the end user as the requesting IP address._

**Comment by andrewsignori-aot** [comment]
@JasonCTang the second AC mentioned "Web" but we believe that it meant "SIMS API", does it make sense?

_Stop logging health check log entries in Web access logs_

**Comment by sslaws** [comment]
@andrewsignori-aot I think it did mean web and I resolved that already but I do see that it applies to the API as well as they are overly verbose.

**Comment by andrewsignori-aot** [comment]
> * Make necessary changes to the web app access logs to write the actual address of the end user as the requesting IP address. (Stephen Laws)
>         
>       
>   
>       Create an issue with the title Make necessary changes to the web app access logs to write the actual address of the end user as the requesting IP address. (Stephen Laws). Press Enter to convert to an issue instantly. Press Alt-Enter to open the create new issue form in the current tab. Press Shift-Enter to open the create new issue form.
> * [ ]  Stop logging health check log entries in Web access logs (probably should be targeting API, right?).
>         
>       
>   
>       Create an issue with the title Stop logging health check log entries in Web access logs (probably should be targeting API, right?).. Press Enter to convert to an issue instantly. Press Alt-Enter to open the create new issue form in the current tab. Press Shift-Enter to open the create new issue form.
> * [ ]  Write the IP Address and GUID (same saved to the `sims.users.user_name`) of the user associated with any API call log entries.

yes @sslaws the APIs ones should be removed. Since the web part is handled, can we adjust the ACs as below?

- [ ] Stop logging health check log entries in SIM API.
- [ ] Write the IP Address and GUID (same saved to the sims.users.user_name) of the user associated with any API call log entries.

**Comment by andrewsignori-aot** [comment]
As discussed with @JasonCTang we are removing the below ACs related to the Web POD that are already implemented.
- [ ] Make necessary changes to the web app access logs to write the actual address of the end user as the requesting IP address. (Stephen Laws)
- [ ] Stop logging health check log entries in Web access logs (probably should be targeting API, right?).

**Comment by AnnaPBashkatova** [comment]
Retagging to 2.2. Milestone to better estimate how much work is outstanding before FT asset release.
